modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java

```diff
@@ -82,13 +103,32 @@ public Pipe(IOControl producerIoControl, ControlledByteBuffer buffer,
         this.buffer = buffer;
         this.name += "_" + name;
         this.baseConfig = baseConfig;
+        this.socketTimeOut = getSocketTimeOut(baseConfig);
     }
 
     public Pipe(ControlledByteBuffer buffer, String name, BaseConfiguration baseConfig) {
         this.buffer = buffer;
         this.name += "_" + name;
         this.baseConfig = baseConfig;
         this.hasHttpProducer = false;
+        this.socketTimeOut = getSocketTimeOut(baseConfig);
+    }
+
+    /**
+     * Extract socket time out value from baseConfig and return default value if not found
+     *
+     * @param baseConfig
+     * @return
+     */
+    private long getSocketTimeOut(BaseConfiguration baseConfig) {
+        if (baseConfig instanceof SourceConfiguration) {
+            return ((SourceConfiguration) baseConfig).getHttpParams()
```

Comment:
Better make sure "((SourceConfiguration) baseConfig).getHttpParams()" cannot be null when it comes to this.